lib/dust.js

```diff
@@ -878,9 +878,12 @@
       try {
         ret = dust.helpers[name](chunk, context, bodies, params);
         if (dust.isThenable(ret)) {
-          return this.await(ret, context, bodies);
+          return chunk.await(ret, context, bodies);
         }
-        return ret;
+        if (ret instanceof Chunk) {
+          return ret;
+        }
+        return chunk.reference(ret, context, 'h');
```

Comment:
Let's have this accept a filters param from the helper and pass it to chunk.reference.  'h' will still be the default though.

Otherwise, looks good!